### PR TITLE
Add metrics in direct memif case

### LINKED ISSUE #360: Metrics server doesn't collect metrics in directmemif case.
Now `metricsServer` doesn't collect metrics if vppconfig has no interfaces, which may be result of case, when client and endpoint chose direcmemif mechanism.

### PR DESCRIPTION
**Pull request by tigrulya-exe**
Should fix #360 

1. Added logic to collect metrics in proxy.
2. Updated directmemif server.
3. Added unit test to check if directmemif server saved metrics in connection path segment.

### CODE REVIEW

File: pkg/networkservice/mechanisms/directmemif/metrics/collector.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+type DirectMemifMetrics struct {
+	// RxBytes is a total number of bytes received from target
+	RxBytes uint
+	// TxBytes is a total number of bytes transmitted to target
+	TxBytes uint
+}
+
+// Collector aggregates metrics
+type Collector interface {
+	Update(interface{}) error
```
Comment (haiodo) [inline pkg/networkservice/mechanisms/directmemif/metrics/collector.go:36]: This looks a bit unclear, how customer of this interface should know how to use it?

Comment (haiodo) [reply pkg/networkservice/mechanisms/directmemif/metrics/collector.go:36]: I think you should define a clear contract for Collector with passing a set of variables, and expect result.

File: pkg/networkservice/mechanisms/directmemif/metrics/collector.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// NewDirectMemifCollector creates directMemifCollector
+func NewDirectMemifCollector() Collector {
+	return &directMemifCollector{}
+}
+
+// Update updates directMemifCollector's metrics with incoming DirectMemifMetrics
+func (d *directMemifCollector) Update(incomingMetrics interface{}) error {
+	memifMetrics, ok := incomingMetrics.(DirectMemifMetrics)
```
Comment (haiodo) [inline pkg/networkservice/mechanisms/directmemif/metrics/collector.go:53]: this is not good to accept interface and downgrade.

File: pkg/networkservice/mechanisms/directmemif/metrics/collector.go (hunk 1 above)
Comment (tigrulya-exe) [reply pkg/networkservice/mechanisms/directmemif/metrics/collector.go]: Replaced interface{} with map of metrics' parameters

File: pkg/networkservice/mechanisms/directmemif/metrics/collector.go (hunk 2 above)
Comment (tigrulya-exe) [reply pkg/networkservice/mechanisms/directmemif/metrics/collector.go]: Fixed.

File: pkg/networkservice/mechanisms/directmemif/metrics/collector.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+// Collector aggregates metrics
+type Collector interface {
+	Update(map[string]string) error
+	Metrics() map[string]string
+}
+
+// Collector aggregates direct memif metrics
+type directMemifCollector struct {
```
Comment (denis-tingajkin) [inline pkg/networkservice/mechanisms/directmemif/metrics/collector.go:42]: I do not think that we actually need to create a separate entity for collecting metrics in directmemif. 

I would suggest to just add new method for proxy
```
type Proxy interface {
   Start() error
   End() error
   Metrics() map[string]string
}
```

And just use this in directmemif server

```
path.GetPathSegments()[path.GetIndex()].Metrics = d.proxies[connectionID].Metrics()
```

**Review by denis-tingajkin — CHANGES_REQUESTED** [review]
I

File: pkg/networkservice/mechanisms/directmemif/metrics/collector.go (hunk 3 above)
Comment (tigrulya-exe) [reply pkg/networkservice/mechanisms/directmemif/metrics/collector.go]: Collector interface removed. Added `Metrics()` method to proxy.

**Review by denis-tingajkin — APPROVED** [review]

**Pull request merged**